### ISSUE TITLE
Remove duplication: create async methods

### ISSUE THREAD

**Issue by mroderick** [sinonjs/sinon#1919]
While working on the codebase tonight, I discovered some unnecessary duplication:

https://github.com/sinonjs/sinon/blob/master/lib/sinon/default-behaviors.js#L266-L272

```js
// create asynchronous versions of callsArg* and yields* methods
forEach(Object.keys(module.exports), function (method) {
    // need to avoid creating anotherasync versions of the newly added async methods
    if (method.match(/^(callsArg|yields)/) && !method.match(/Async/)) {
        module.exports[method + "Async"] = createAsyncVersion(method);
    }
});
```

and

https://github.com/sinonjs/sinon/blob/master/lib/sinon/behavior.js#L216-L222

```js
// create asynchronous versions of callsArg* and yields* methods
forEach(Object.keys(proto), function (method) {
    // need to avoid creating anotherasync versions of the newly added async methods
    if (method.match(/^(callsArg|yields)/) && !method.match(/Async/)) {
        proto[method + "Async"] = createAsyncVersion(method);
    }
});
```

We should refactor the code to only have one copy of this, and re-use it.

**Comment by LEQADA** [comment]
@mroderick I'd like to pick up this task. Any suggestions on where to put a new util function?

**Comment by mroderick** [comment]
@LEQADA that would be great 👍 

I think you could create a file in [`lib/sinon/util/core`](https://github.com/sinonjs/sinon/tree/master/lib/sinon/util/core)

**Comment by morgsmccauley** [comment]
@LEQADA did you make any progress on this? I'd like to pick this up if it's not being worked on 😄

**Comment by LEQADA** [comment]
@morgsmccauley sorry for a long inactivity. I already started to work on the issue but didn't have time to write tests. I plan to finish it next week if it's not urgent. If it's urgent then sure take it over. 
Otherwise I'd ask to give me some more days. :slightly_smiling_face:

**Comment by mroderick** [comment]
There are plenty [more issues across the sinonjs repositories with "Help wanted" label](https://github.com/search?q=org%3Asinonjs+is%3Aopen+label%3A%22Help+wanted%22&unscoped_q=is%3Aopen+label%3A%22Help+wanted%22).

**Comment by mroderick** [comment]
LIke this one to find a solution for an unmaintained dependency: https://github.com/sinonjs/nise/issues/66

**Comment by morgsmccauley** [comment]
Great, thanks @mroderick!

**Comment by mroderick** [comment]
Fixed with #1942
